superset/assets/visualizations/nvd3_vis.js

```diff
@@ -317,45 +316,33 @@ function nvd3Vis(slice, payload) {
     if (fd.x_log_scale) {
       chart.xScale(d3.scale.log());
     }
-    let xAxisFormatter;
-    if (vizType === 'bubble') {
-      xAxisFormatter = d3.format('.3s');
-    } else if (fd.x_axis_format === 'smart_date') {
-      xAxisFormatter = formatDate;
-      chart.xAxis.tickFormat(xAxisFormatter);
-    } else if (fd.x_axis_format !== undefined) {
-      xAxisFormatter = timeFormatFactory(fd.x_axis_format);
-      chart.xAxis.tickFormat(xAxisFormatter);
-    }
 
     const isTimeSeries = timeStampFormats.indexOf(fd.x_axis_format) > -1;
     // if x axis format is a date format, rotate label 90 degrees
     if (isTimeSeries) {
       chart.xAxis.rotateLabels(45);
     }
 
+    let xAxisFormatter = d3FormatPreset(fd.x_axis_format);
+    if (isTimeSeries && fd.x_axis_format) {
+      xAxisFormatter = d3TimeFormatPreset(fd.x_axis_format);
+    }
     if (chart.hasOwnProperty('x2Axis')) {
       chart.x2Axis.tickFormat(xAxisFormatter);
       height += 30;
     }
+    if (chart.hasOwnProperty('xAxis')) {
+      chart.xAxis.tickFormat(xAxisFormatter);
+    }
 
-    if (vizType === 'bubble') {
-      chart.xAxis.tickFormat(d3.format('.3s'));
-    } else if (fd.x_axis_format === 'smart_date') {
-      chart.xAxis.tickFormat(formatDate);
-    } else if (fd.x_axis_format !== undefined) {
-      chart.xAxis.tickFormat(timeFormatFactory(fd.x_axis_format));
+    const yAxisFormatter = d3FormatPreset(fd.y_axis_format);
+    if (chart.hasOwnProperty('yAxis')) {
+      chart.yAxis.tickFormat(yAxisFormatter);
     }
-    if (chart.yAxis !== undefined) {
-      chart.yAxis.tickFormat(d3.format('.3s'));
+    if (chart.hasOwnProperty('y2Axis')) {
```

Comment:
i think `if (chart.yAxis && chart.yAxis tickFormat)` would be less brittle here. 
 
`chart.hasOwnProperty('y2Axis')` would return `true` even if the value is `undefined`, in which case `chart.y2Axis.tickFormat()` would throw an error. do we always know that `chart` won't have the key with a value of false or undefined?

![screenshot 2017-06-12 23 30 53](https://user-images.githubusercontent.com/130878/27069033-cacae98a-4fc7-11e7-8bee-9e37a6682a67.png)